### PR TITLE
[SPACES] Support file and folders operations with Spaces in DocumentsProvider 

### PR DESCRIPTION
**Pull request by JuancaG05**
## Related Issues
App: https://github.com/owncloud/android/issues/3881
_____

### CODE REVIEW

**Comment by jesmrec** [comment]
One open question here:

"Move to other space in same account" operation is banned in the app, but available in doc provider. 

i guess that it's not straightforward to prevent this... but it is a shortcut to jump over the app restriction. Is this OK?

CC @michaelstingl

**Comment by JuancaG05** [comment]
In Android, since we use the native documents provider, we cannot restrict this behaviour.

In fact, the move operation is not performed as it is, since we invoke the move usecase and an exception is thrown if we try to move to a different space. When this happens, the documents provider catches internally the exception and performs an "artificial" move, which means downloading the file from the source, upload it to the destination and deleting it from the source. This artificial move is performed as well when trying to move between accounts through the documents provider -> not allowed in the app but from the document provider.

In conclusion, a remote move is not performed in server, but we cannot restrict it since it is a native and internal behaviour from Android. We'll see if this is doable in future versions of documents provider.

/CC @michaelstingl

**Comment by jesmrec** [comment]
A complete sanity test was done over the feature, including operations over oCIS accounts with spaces, oC10 accounts and other providers.

Some open issues are still reproducible: 

https://github.com/owncloud/android/issues/3675
https://github.com/owncloud/android/issues/2659
https://github.com/owncloud/android/issues/2658

These three, related with `Copy` and `Move` operations (read, upload and delete under the hood) are still there. The most unstable scenario is copying/moving tree structures with more than 2-deep levels. Sometimes works, sometimes fails, sometimes is only copied or moved a part of the files. If the operation is done over two different oC accounts/providers (no matter if oC10 or oCIS), it's unstable even with a few amount of items.  My guess: some internal memory restrictions (did not find anything in [official documentation](https://developer.android.com/guide/topics/providers/document-provider?hl=en) about this)

Keep in mind also this issue with improvements that point to the mentioned issues: https://github.com/owncloud/android/issues/2336

Since those are known issues and do not affect the oCIS' spaces implementation, i do not have objections to move this forward, keeping in mind that some scenarios are not working fine. If any other idea/suggestion/solution to improve the known bad scenarios, is totally welcome.
